### ISSUE TITLE
Strage behavior on release PR

### ISSUE THREAD

**Issue by sassman** [MarcoIeni/release-plz#582]
## Bug description

I have observed a bit of strange behavior that I cannot explain really.
On my project curlz I have for example [one release PR](https://github.com/curlz-rs/curlz/pull/34)
created by release-plz. That looks good and is all well.

However, I've noted the 2 things:

## Builds are not triggered
When the PR is updated / created no build is triggered. 
But when I take the branch, and push it myself with a different name ([see PR](https://github.com/curlz-rs/curlz/pull/36)) then the build pipeline is triggered as normally expected.

I suspect it has something todo with the user context that is used to open the PR. It is maybe not the same as by the provided `GITHUB_TOKEN`.

## Merging the release PR, pushes the tag but without triggering any pipeline
In this very repo I have a dedicated pipeline for when a tag is pushed (a cargo publish pipeline).
This is always triggered if I push the tag myself.

However, when release-plz is pushing the tag, just nothing happens, no pipeline is triggered.

I must assume the root cause is similar to the first issue, somehow related to the user context used when pushing the tag.

- Would you like to work on a fix? [y] 
I checked already the code that the `GITHUB_TOKEN` is used correctly, the only place that I didn't yet check is the API interaction for opening a PR and pushing a tag.

Not sure if I could solve it in a timely manner.

## To Reproduce

Very hard to reproduce, I would believe that one can only reproduce it on repos that belong to an org not to a personal account.

But the workflow is quite simple, change code, let the release-plz create the PR, observe that no pipeline is triggered.

## Expected behavior

I would expect the pipeline is triggered as if I would create the PR or push the tag myself.

## Environment

- OS: Ubuntu 22.04
- release-plz version: using the action that depends on main

## Additional context

here is the PR from above again:
https://github.com/curlz-rs/curlz/pull/36

**Comment by MarcoIeni** [comment]
Please read the "[Triggering further workflow runs](https://marcoieni.github.io/release-plz/github-action.html#triggering-further-workflow-runs)" section of the docs.
[This](https://github.com/MarcoIeni/release-plz/blob/d14ee7ebf227ba9f3a2e694af38c1fc8b475841a/.github/workflows/release-plz.yml#L20) is how I solved it in release-plz itself.

**Comment by sassman** [comment]
Thanks for pointing this out. 

But I did create a dedicated personal access token. So the default token is not involved.

**Comment by sassman** [comment]
But the workflow permissions were not the same as in the documentation. So maybe this already solves the issue. 

Thanks a lot.

**Comment by MarcoIeni** [comment]
I'm not sure if the permissions will fix your issue.
Please, reopen this issue if they don't.
This is a confusing behavior from release-plz, so I would like to fix it, either by editing the code or by improving the docs :)

**Comment by MarcoIeni** [comment]
btw I was having a look at curlz. 
I think you can just remove the file [https://github.com/curlz-rs/curlz/blob/main/.github/workflows/release.yml](release.yml) because that's already done by release-plz 🤔 

If you want to keep the release separate, please update your release-plz.yml file to only do a release-pr. See the [release-pr](https://marcoieni.github.io/release-plz/github-action.html#example-release-pr) example

**Comment by sassman** [comment]
> I think you can just remove the file [https://github.com/curlz-rs/curlz/blob/main/.github/workflows/release.yml](https://github.com/MarcoIeni/release-plz/issues/release.yml) because that's already done by release-plz 🤔

Yeah, I have realized it too and was also not sure if I want to give away the process.
But I'll give it a shot now and see how that rolls.

I figured out that I have used one of the new GH Tokens (the beta feature) and that this seems to have problems. Using now a classic token and the PR has a build and all seems to behave.

Thanks again for your support Marco
